### PR TITLE
[SPIKE] - Proposed API Changes for Pinning Video Tiles in Video Gallery

### PR DESCRIPTION
**Pull request by anjulgarg**
# What
[SPIKE] - Proposed API Changes for Pinning Video Tiles in Video Gallery

The other part of this change
https://github.com/Azure/communication-ui-library/pull/2365

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by JamesBurnside — COMMENTED** [review]
LGTM!

Leaving nit comments here to not comment the changelist for others reviewing:
* click is normally the suffix in our codebase, so `onPinIconClick` or just `onPinClick`
* Typically we have a single function for onX and offX (e.g. [toggle button](https://developer.microsoft.com/en-us/fluentui#/controls/web/toggle) has just onChange for both on and off) so for `onPinParticipant` and `onUnpinParticipant` RemoteVideoTile.tsx layer I would suggest just have `onPinChange: (newPinState: 'on' | 'off') => void`

File: packages/react-components/src/components/RemoteVideoTile.tsx [hunk 1]
```diff
@@ -111,6 +114,20 @@ export const _RemoteVideoTile = React.memo(
         /* @conditional-compile-remove(one-to-n-calling) */
         /* @conditional-compile-remove(PSTN-calls) */
         participantState={props.participantState}
+        isPinned={props.isPinned}
+        onClickPinIcon={() => {
+          props.onUnpinParticipant?.();
+        }}
+        menuItems={[
```
Comment (JamesBurnside) [inline packages/react-components/src/components/RemoteVideoTile.tsx:121]: if we don't add these conditionally its kinda a breaking change - Contoso will now have a pin button that doesn't work.
Consider only including the menu items if `onPinParticipant` is provided?

Comment (anjulgarg) [inline packages/react-components/src/components/RemoteVideoTile.tsx:121]: Open Question:
How do we make this work with the drawer menu on mobile?
Should we refrain from using a Fluent type here and introduce a simplified structure that works with both the contextual menu and the drawer menu?

File: packages/react-components/src/components/VideoTile.tsx [hunk 2]
```diff
@@ -125,6 +125,25 @@ export interface VideoTileProps {
...
+   * The pin icon is only visible when `isPinned` is true.
+   */
+  onClickPinIcon?: () => void;
+  /* @conditional-compile-remove(pinned-participants) */
+  /**
+   * Display custom menu items in the VideoTile's contextual menu.
+   * It uses Fluent UI's ContextualMenu component. Thus the menu items should be of type `IContextualMenuItem`.
+   * For more information see https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu
+   */
+  menuItems?: IContextualMenuItem[];
```
Comment (anjulgarg) [inline packages/react-components/src/components/VideoTile.tsx:146]: Open Question:
How do we make this work with the drawer menu on mobile?
Should we refrain from using a Fluent type here and introduce a simplified structure that works with both the contextual menu and the drawer menu?

**Pull request closed without merging**